src/packages/client/scripts/postinstall.js

```diff
@@ -191,3 +200,62 @@ async function makeDir(input) {
 
   return make(path.resolve(input))
 }
+
+/**
+ * Get the command that triggered this postinstall script being run. If there is
+ * an error while attempting to get this value then the string constant
+ * 'ERROR_WHILE_FINDING_POSTINSTALL_TRIGGER' is returned.
+ */
+function getPostInstallTrigger() {
+  /*
+  npm_config_argv` is not officially documented so here are our research notes 
+
+  `npm_config_argv` is available to the postinstall script when the containing package has been installed by npm into some project.
+
+  An example of its value:
+
+  ```
+  npm_config_argv: '{"remain":["../test"],"cooked":["add","../test"],"original":["add","../test"]}',
+  ```
+
+  We are interesting in the data contained in the "original" field.
+
+  Trivia/Note: `npm_config_argv` is not available when running e.g. `npm install` on the containing package itself (e.g. when working on it)
+
+  Yarn mimics this data and environment variable. Here is an example following `yarn add` for the same package:
+
+  ```
+  npm_config_argv: '{"remain":[],"cooked":["add"],"original":["add","../test"]}'
+  ```
+
+  Other package managers like `pnpm` have not been tested.
+  */
+
+  const ERROR_WHILE_FINDING_POSTINSTALL_TRIGGER =
+    'ERROR_WHILE_FINDING_POSTINSTALL_TRIGGER'
+
+  const maybe_npm_config_argv_string = process.env.npm_config_argv
+
+  if (maybe_npm_config_argv_string === undefined) {
+    return ERROR_WHILE_FINDING_POSTINSTALL_TRIGGER
```

Comment:
This one is not really an error from my point of view, just not defined

Reply:
Is it possible for postinstall to run without it being defined?

Reply:
Well you could do `node postinstall.js` manually for example, or maybe some package manager will not define it.

Reply:
I suppose the important thing is to identify these cases explicitly. Any name proposals? A fairly general one might be `UNABLE_TO_ESTABLISH_POSTINSTALL_TRIGGER`.

Reply:
If we start seeing a lot of these cases we might start wondering WHY and then want to treat each "failure" case with its own keyword. Example: `UNABLE_TO_ESTABLISH_POSTINSTALL_TRIGGER__NO_ENVAR`.

Reply:
Ok I took the approach of just labeling all the cases explicitly. Simple/factual approach now. Feel free to re-open discussion if you can still see an issue!